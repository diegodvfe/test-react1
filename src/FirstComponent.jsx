import PropTypes  from 'prop-types'

// const GetSaludos = {
//     message: 'Hola'
// };

// const GetName = 'Vanessa!!';

// const getResults = (a, c) => {
//     return a + c;
// };

export const FirstComponent = ({title, subTitle, name}) => {

        
    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    );
};


FirstComponent.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired
}

FirstComponent.defaultProps = {
    // title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Diego Lopez'
}