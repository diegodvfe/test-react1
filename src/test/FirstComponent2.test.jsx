import { render, screen } from '@testing-library/react';
import { FirstComponent } from '../FirstComponent';

describe('Este es una prueba de factorizacion de la aplicacion', () => {
    const title = 'Hola, soy Diego';
    const subTitle = 'Soy un subtitulo';
    const name = ' Me llamo React';
    // console.log(title)

    test('Este debe de regresar un snapshot del componente', () => {
        const { container } = render(<FirstComponent title={title} />);

        expect(container).toMatchSnapshot();
    });

    test('Este debe de evaluar el titulo Hola soy Diego', () => {
        render(<FirstComponent title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe de evaluar el titulo en un h1', () => {
        render(<FirstComponent title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render(<FirstComponent title={title} subTitle={subTitle} />);

        expect(screen.getAllByText(subTitle).length).toBe(2);
    });

    test('debe de evaluar el prop de name en la aplicacion', () => {
        render(<FirstComponent name={name} />);

        expect(screen.getAllByText(name).length).toBe(1);
    });
});
