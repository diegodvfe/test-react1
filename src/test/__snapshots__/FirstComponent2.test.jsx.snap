// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Este es una prueba de factorizacion de la aplicacion Este debe de regresar un snapshot del componente 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Diego
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Diego Lopez
  </p>
</div>
`;
